src/create-support-ticket.ts

```diff
@@ -0,0 +1,201 @@
+  const { AWS_REGION } = process.env;
+  const client = new SQSClient({ region: AWS_REGION });
+  const message: SendMessageRequest = {
+    QueueUrl: queueUrl,
+    MessageBody: messageBody,
+  };
+  return client.send(new SendMessageCommand(message));
+};
+
+export const handler = async (event: SNSEvent): Promise<void> => {
+  const {
+    ZENDESK_GROUP_ID_KEY,
+    ZENDESK_TAGS_KEY,
+    ZENDESK_API_TOKEN_KEY,
+    ZENDESK_API_USER_KEY,
+    ZENDESK_API_URL_KEY,
+    DLQ_URL,
+  } = process.env;
+  await Promise.all(
+    event.Records.map(async (record) => {
+      try {
+        const receivedEvent: SuspiciousActivityEvent = JSON.parse(
+          record.Sns.Message
+        );
+        if (
+          !ZENDESK_API_USER_KEY ||
+          !ZENDESK_API_TOKEN_KEY ||
+          !ZENDESK_API_URL_KEY ||
+          !ZENDESK_GROUP_ID_KEY
+        ) {
+          throw new Error(
+            "Not all environment variables required to successfully send to Zendesk are provided."
+          );
+        }
+
+        validateSuspiciousActivity(receivedEvent);
+        const zendeskUserName = await getSecret(ZENDESK_API_USER_KEY, {
+          maxAge: 900,
+        });
+        const zendeskUserPassword = await getSecret(ZENDESK_API_TOKEN_KEY, {
+          maxAge: 900,
+        });
+        const zendeskGroupId = await getSecret(ZENDESK_GROUP_ID_KEY, {
+          maxAge: 900,
+        });
+        const zendeskURL = await getSecret(ZENDESK_API_URL_KEY, {
+          maxAge: 900,
+        });
+        const zendeskTicketTags = ZENDESK_TAGS_KEY
+          ? await getSecret(ZENDESK_TAGS_KEY, {
+              maxAge: 900,
+            })
+          : undefined;
+
+        if (
+          !zendeskUserName ||
+          !zendeskUserPassword ||
+          !zendeskGroupId ||
+          !zendeskURL
+        ) {
+          throw new Error("Required zendesk secrets not configured");
+        }
+
+        const tags = (zendeskTicketTags as string)?.split(",");
+        const ticket: CreateTicketPayload = {
+          ticket: {
+            subject: "REPORT_SUSPICIOUS_ACTIVITY -OLH TEST IGNORE",
+            comment: {
+              html_body: formatCommentBody(receivedEvent),
+            },
+            group_id: Number(zendeskGroupId),
+            tags,
+          },
+        };
+
+        await createTicket(
+          ticket,
+          zendeskURL as string,
+          zendeskUserName as string,
+          zendeskUserPassword as string
+        );
+      } catch (error: unknown) {
+        console.error(
+          `[Error occurred], unable to send suspicious activity event to Zendesk, ${
+            (error as Error).message
+          }`
+        );
+        const { AWS_REGION } = process.env;
+        const client = new SQSClient({ region: AWS_REGION });
+        const message: SendMessageRequest = {
+          QueueUrl: DLQ_URL,
+          MessageBody: record.Sns.Message,
+        };
+        const result = await client.send(new SendMessageCommand(message));
```

Comment:
If this fails, then we won't know what event caused the failure. It might be worth printing out message (redacted) in the console.error on line 183.

Reply:
Agreed, now adding the event identifier as part of the intial log